lang/mrustc/Portfile

```diff
@@ -59,46 +62,120 @@ compiler.blacklist *gcc*
 #}
 
-# https://github.com/thepowersgang/mrustc/pull/192
-patchfiles-append patch-mrustc-macos-bootstrap.diff
+# Patches avaialble as git branch: https://github.com/catap/mrustc/tree/macports
+patch.pre_args  -p1
+patchfiles      0001-1.39.0-Fixes-for-bootstrapping-1.39-on-macOS.patch \
+                0002-Added-LINKFLAGS_EXTRA.patch \
+                0003-run_rustc-respect-j-option.patch \
+                0004-Added-LLVM_CMAKE_OPTS_EXTRA.patch \
+                0005-Let-MacPorts-set-a-default-value-for-MRUSTC_LIBDIR.patch \
+                0006-Let-MacPorts-define-the-default-compiler.patch \
+                0007-Let-MacPorts-customize-the-default-Rust-target-versi.patch
 
-# Set the default MRUSTC_LIBDIR
-patchfiles-append patch-mrustc-macports-libdir.diff
 post-patch {
+    # Set the default MRUSTC_LIBDIR
     reinplace "s|@MACPORTS_MRUSTC_LIBDIR@|${prefix}/lib/mrustc-${rust_version_major}|" ${worksrcpath}/src/main.cpp
-}
-
-# Set the default CC
-patchfiles-append patch-mrustc-macports-compiler.diff
-post-patch {
+    # Set the default CC
     reinplace "s|@MACPORTS_MRUSTC_CC@|${configure.cc}|" ${worksrcpath}/src/trans/codegen_c.cpp
-}
-
-# Set the default gTargetVersion
-patchfiles-append patch-mrustc-macports-version.diff
-post-patch {
+    # Set the default gTargetVersion
     reinplace "s|@MACPORTS_MRUSTC_TARGET_VER@|${rust_version_major_underscore}|" ${worksrcpath}/src/main.cpp
-}
 
-# Patch the Rust source code with the patch included with mrustc
-post-patch {
+    # Patch the Rust source code with the patch included with mrustc
     system -W ${workpath}/rustc-${rust_version}-src "patch -p0 < ${worksrcpath}/rustc-${rust_version}-src.patch"
     file rename ${workpath}/rustc-${rust_version}-src ${worksrcpath}
+    # prevent it from re-download src
+    touch ${worksrcpath}/rustc-${rust_version}-src/dl-version
 }
 
-build {
-    system -W ${worksrcpath} "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} CXXFLAGS_EXTRA='${configure.cxxflags}' ${build.cmd} all"
-    system -W ${worksrcpath}/tools/minicargo "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} ${build.cmd} all"
-    system -W ${worksrcpath} "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} RUSTC_VERSION=${rust_version} RUSTC_TARGET=${build_arch}-apple-darwin MRUSTC_LIBDIR=output-${rust_version} MRUSTC_TARGET_VER=${rust_version_major} ${build.cmd} -f minicargo.mk LIBS"
+depends_build-append \
+                port:cmake \
+                port:git \
+                port:libtool \
+                port:pkgconfig
+
+depends_lib-append \
+                port:curl \
+                port:libgit2 \
+                port:nghttp2 \
+                port:zlib
+
+# use the system python27 if present
+if {${os.platform} eq "darwin" && ${os.major} >= 11} {
+    set pythonfullpath   /usr/bin/python2.7
+} else {
+    set pythonfullpath   ${prefix}/bin/python2.7
+    depends_lib-append   port:python27
```

Comment:
Should this be `depends_run`? (`depends_run` is used if there is no library linkage.)

Reply:
In llvm/clang, it is listed as depends_lib, because you need it for both build and run (depends_build + depends_run = depends_lib). Here you only need it for the build I believe, and not at runtime, so it could be depends_build.

Reply:
@evanmiller nope, the only reason to use python2.7 is make clang's cmake happy. After it is built, python2.7 is useless.

Reply:
depends_build, as I mentioned